### DISCUSSION TITLE
install acebase-server

### DISCUSSION THREAD

**Discussion by dips-programaya** [appy-one/acebase#228] [Q&A] [answered]
hi

I'm trying to install acebase-server on a hosting from cpanel.

I follow the steps:
create a domain
in the domain folder I run npm install acebase-server
cd node_modules/acebase-server
npm run test

and gives the following error:
> acebase-server@1.18.0 test /home/programa/aplicaciones/acebase/node_modules/acebase-server
> set NODE_ENV=development && npm run start

> acebase-server@1.18.0 start /home/programa/aplicaciones/acebase/node_modules/acebase-server
> node dist/esm/start.js DBNAME=default DBPATH=. HOST=localhost PORT=3000 TXLOG=1 AUTH=1

/home/programa/aplicaciones/acebase/node_modules/acebase-server/dist/esm/start.js:1
(function (exports, require, module, __filename, __dirname) { import { AceBaseServer } from './server.js';
                                                              ^^^^^^

SyntaxError: Unexpected token import
    at new Script (vm.js:51:7)
    at createScript (vm.js:136:10)
    at Object.runInThisContext (vm.js:197:10)
    at Module._compile (internal/modules/cjs/loader.js:618:28)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:665:10)
    at Module.load (internal/modules/cjs/loader.js:566:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:506:12)
    at Function.Module._load (internal/modules/cjs/loader.js:498:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:695:10)
    at startup (internal/bootstrap/node.js:201:19)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! acebase-server@1.18.0 start: `node dist/esm/start.js DBNAME=default DBPATH=. HOST=localhost PORT=3000 TXLOG=1 AUTH=1`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the acebase-server@1.18.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/programa/.npm/_logs/2023-05-22T22_06_12_439Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! acebase-server@1.18.0 test: `set NODE_ENV=development && npm run start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the acebase-server@1.18.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/programa/.npm/_logs/2023-05-22T22_06_12_451Z-debug.log

what am I doing wrong?
And then how do I continue?

ruben

**Comment by appy-one** [comment, marked as answer]
To run an AceBase server using the ES Modules build, you need at least Node.js 14. If you are using an older version (but at least 10.4), you can start it using the CommonJS build with `node dist/cjs/start.js ...`

**Comment by dips-programaya** [comment]
thank you
